tests/feature-store/test_steps.py

```diff
@@ -56,3 +72,363 @@ def test_set_event_random_id():
     resp = server.test(body={"data": "123"}, event_id="XYZ")
     server.wait_for_completion()
     assert resp["id"] != "XYZ", "id was not overwritten"
+
+
+def test_pandas_step_onehot(rundb_mock):
+    data = get_data()
+    # One Hot Encode the newly defined mappings
+    one_hot_encoder_mapping = {
+        "department": list(data["department"].unique()),
+    }
+    # Define the corresponding FeatureSet
+    data_set_pandas = fs.FeatureSet(
+        "fs-new",
+        entities=[fs.Entity("id")],
+        description="feature set",
+        engine="pandas",
+    )
+    # Pre-processing grpah steps
+    data_set_pandas.graph.to(OneHotEncoder(mapping=one_hot_encoder_mapping))
+    data_set_pandas._run_db = rundb_mock
+
+    data_set_pandas.reload = unittest.mock.Mock()
+    data_set_pandas.save = unittest.mock.Mock()
+    data_set_pandas.purge_targets = unittest.mock.Mock()
+
+    # Ingest our transactions dataset through our defined pipeline
+    df_pandas = fs.ingest(
+        data_set_pandas, data, targets=[ParquetTarget(path="./temp.parquet")]
+    )
+
+    assert isinstance(df_pandas, pd.DataFrame)
```

Comment:
Should assert the expected result, not just the type.

Reply:
In all tests of course.

Reply:
you can see that in the next lines I compared the pandas engine result to the storey engine result.

Reply:
Right, but you could also assert the expected result, and that would be better.